modular_zubbers/code/modules/status_indicators/status_indicator.dm

```diff
@@ -120,54 +112,33 @@ GLOBAL_LIST_INIT(potential_indicators, list(
 	SIGNAL_HANDLER
 	is_confused() ? add_status_indicator(CONFUSED) : remove_status_indicator(CONFUSED)
 
-/// Adds a status indicator to the mob. Takes an image as an argument. If it exists, it won't dupe it.
-/datum/component/status_indicator/proc/add_status_indicator(image/prospective_indicator)
-	if(get_status_indicator(prospective_indicator)) // No duplicates, please.
-		return
-
-	prospective_indicator = GLOB.potential_indicators[prospective_indicator]
-	prospective_indicator.loc = src
-	LAZYADD(status_indicators, prospective_indicator)
-	handle_status_indicators(prospective_indicator)
+/// Adds a status indicator to the mob. If it exists, it won't dupe it.
+/datum/component/status_indicator/proc/add_status_indicator(prospective_indicator)
+	var/obj/effect/overlay/status_indicator/this_indicator
+	if(!status_indicators[prospective_indicator])
+		this_indicator = new
+		this_indicator.icon_state = prospective_indicator
+		status_indicators[prospective_indicator] = this_indicator
+		animate_new_indicator(this_indicator)
 
 /// Similar to add_status_indicator() but removes it instead, and nulls the list if it becomes empty as a result.
-/datum/component/status_indicator/proc/remove_status_indicator(image/prospective_indicator)
-	prospective_indicator = get_status_indicator(prospective_indicator)
-
-	attached_mob.cut_overlay(prospective_indicator)
-	LAZYREMOVE(status_indicators, prospective_indicator)
-	handle_status_indicators(prospective_indicator)
-
-/// Finds a status indicator on a mob.
-/datum/component/status_indicator/proc/get_status_indicator(image/prospective_indicator)
-
-	for(var/image/indicator in status_indicators)
-		if(indicator.icon_state == prospective_indicator)
-			return indicator
-	return LAZYACCESS(status_indicators, LAZYFIND(status_indicators, prospective_indicator))
-
-/// Cuts all the indicators on a mob in a loop.
-/datum/component/status_indicator/proc/cut_indicators_overlays()
-	SIGNAL_HANDLER
-	for(var/prospective_indicator in status_indicators)
-		attached_mob.cut_overlay(prospective_indicator)
+/datum/component/status_indicator/proc/remove_status_indicator(prospective_indicator)
+	var/obj/effect/overlay/status_indicator/resolved_indicator = status_indicators[prospective_indicator]
+	if(resolved_indicator)
+		status_indicators.Remove(prospective_indicator, resolved_indicator)
```

Comment:
```suggestion
		status_indicators[prospective_indicator] = null
```
the .Remove list proc just takes a comma separated list of elements to remove, this doesn't remove it in an associative way, setting an associative list key's value to null is effectively removing it